auction/contracts/AuctionHouse.sol

```diff
@@ -0,0 +1,333 @@
+        doTransfers(_auctionId);
+        deactivateAuction(_auctionId);
+    }
+
+    /// @dev transfers auction assets (sellAsset to buyer, buyAsset to seller)
+    function doTransfers(uint _auctionId) internal {
+        Auction memory currentAuction = auctions[_auctionId];
+        address seller = currentAuction.seller;
+        address buyer = currentAuction.buyer;
+        if (buyer != address(0x0)) {//bid exists
+            LibOrderDataV2.DataV2 memory bidData = LibBidDataV1.getPaymentData(currentAuction.lastBid.data, currentAuction.lastBid.dataType);
+            if (bidData.payouts.length == 0){
+                LibPart.Part[] memory payout = new LibPart.Part[](1);
+                payout[0].account = payable(buyer);
+                payout[0].value = 10000;
+                bidData.payouts = payout;
+            }
+            LibOrderDataV2.DataV2 memory aucData = LibAucDataV1.getPaymentData(currentAuction.data, currentAuction.dataType);
+            if (aucData.payouts.length == 0){
+                LibPart.Part[] memory payout = new LibPart.Part[](1);
+                payout[0].account = payable(seller);
+                payout[0].value = 10000;
+                aucData.payouts = payout;
+            }
+            doTransfersWithFees(
+                currentAuction.lastBid.amount, 
+                address(this), 
+                MatchFees(currentAuction.protocolFee, currentAuction.protocolFee, LibFeeSide.FeeSide.TAKE),
+                bidData, 
+                aucData, 
+                currentAuction.buyAsset, 
+                currentAuction.sellAsset.assetType, 
+                TO_MAKER
+            );
+            transferPayouts(currentAuction.sellAsset.assetType, currentAuction.sellAsset.value, address(this), bidData.payouts, TO_TAKER);
+        } else {
+            transfer(currentAuction.sellAsset, address(this), seller, TO_SELLER, UNLOCK); //nft back to seller
+        }
+    }
+
+    /// @dev returns true if auction started and hasn't finished yet, false otherwise
+    function checkAuctionRangeTime(uint _auctionId) public view returns (bool){
+        uint currentTime = block.timestamp;
+        LibAucDataV1.DataV1 memory aucData = LibAucDataV1.parse(auctions[_auctionId].data, auctions[_auctionId].dataType);
+        if (aucData.startTime > 0 && aucData.startTime > currentTime) {
+            return false;
+        }
+        uint endTime = auctions[_auctionId].endTime;
+        if (endTime > 0 && endTime < currentTime){
+            return false;
+        }
+        
+        return true;
+    }
+
+    /// @dev deletes auction after finalizing
+    function deactivateAuction(uint _auctionId) internal {
+        emit AuctionFinished(_auctionId, auctions[_auctionId]);
+        deleteAuctionForToken(auctions[_auctionId].sellAsset);
+        delete auctions[_auctionId];
+    }
+
+    /// @dev cancels existing auction without bid
+    function cancel(uint _auctionId) external {
```

Comment:
is it ok that 
1 during canceling we have 2 events: canceled and finished
2 in deactivate method we have "finished" event
?
maybe it would be better to create a special method
_finishAuction (maybe with another name)
emit AuctionFinished();
deactivateAuction(_auctionId);
}
delete "emit AuctionFinished" from "deactivateAuction"
call "_finishAuction" instead of "deactivateAuction" anywhere except in "Cancel" method

Reply:
1. yes i was asked by back end team to emit 2 events in that case
2. we should always deactivate auction when we emit auctionFinished
i think there is no need for these changes if we want to to have 2 events in cancel, if i understood you correctly